src/lib.rs

```diff
@@ -23,19 +23,22 @@ use std::path::Path;
 /// A utility function for build scripts to generate bindings for a crate, using
 /// a `cbindgen.toml` if it exists.
 pub fn generate<P: AsRef<Path>>(crate_dir: P) -> Result<Bindings, Error> {
-    let config = Config::from_root_or_default(crate_dir.as_ref());
+    let config = ConfigLoader::from_root_or_default(crate_dir.as_ref());
+    let root = config.root().clone();
 
-    generate_with_config(crate_dir, config)
+    generate_with_config(crate_dir.as_ref(), &root, config.into())
 }
 
 /// A utility function for build scripts to generate bindings for a crate with a
 /// custom config.
 pub fn generate_with_config<P: AsRef<Path>>(
```

Comment:
Can we keep this API unchanged and default to no root when providing an in-memory configuration?

Reply:
I can do that, but then I would make embed panic if an embed was requested but no root was given.